ExampleMod/Content/Tiles/ExampleVanillaConversionTiles.cs

```diff
@@ -0,0 +1,255 @@
+			if (conversionType == BiomeConversionID.Corruption)
+				WorldGen.ConvertTile(i, j, Type);
+			else
+				WorldGen.ConvertTile(i, j, ModContent.TileType<CrimsonFossilTile>());
+			return false;
+		}
+
+		public override bool Convert(int i, int j, int conversionType) {
+			if (conversionType == BiomeConversionID.Purity || conversionType == BiomeConversionID.Sand || conversionType == BiomeConversionID.PurificationPowder) {
+				WorldGen.ConvertTile(i, j, TileID.DesertFossil);
+				return false;
+			}
+			else if (conversionType == BiomeConversionID.Hallow) {
+				WorldGen.ConvertTile(i, j, ModContent.TileType<HallowedFossilTile>());
+				return false;
+			}
+			else if (conversionType == BiomeConversionID.Crimson) {
+				WorldGen.ConvertTile(i, j, ModContent.TileType<CrimsonFossilTile>());
+				return false;
+			}
+
+			return true;
+		}
+
+		public override void RandomUpdate(int i, int j) {
+			//Account for journey mode disabling infections and plantera slowing down biome spread rate
+			if (Main.hardMode && (NPC.downedPlantBoss && WorldGen.genRand.NextBool(2)) || !WorldGen.AllowedToSpreadInfections)
+				return;
+
+			//Check a random nearby tile and try to convert it into hallow
+			int testX = i + WorldGen.genRand.Next(-3, 4);
+			int testY = j + WorldGen.genRand.Next(-3, 4);
+			if (!WorldGen.InWorld(testX, testY, 10))
+				continue;
+
+			//Chlorophyte prevents and repels biome spread, so we should account for that
+			if (WorldGen.nearbyChlorophyte(testX, testY)) {
+				WorldGen.ChlorophyteDefense(testX, testY);
+			}
+			else {
+				//Sunflowers prevent spread
+				if (WorldGen.CountNearBlocksTypes(testX, testY, 2, 1, TileID.Sunflower) > 0)
+					continue;
+
+				WorldGen.Convert(testX, testY, BiomeConversionID.Corruption, 1);
+			}
+		}
+	}
+
+	public class CrimsonFossilTile : ModTile
+	{
+		public override void SetStaticDefaults() {
+			Main.tileSolid[Type] = true;
+			TileID.Sets.Crimson[Type] = true;
+			TileID.Sets.CrimsonBiome[Type] = true;
+			TileID.Sets.CrimsonBiomeSight[Type] = true;
+			TileID.Sets.CrimsonCountCollection[Type] = true;
+			DustType = DustID.Crimstone;
+			AddMapEntry(new Color(112, 33, 32));
+		}
+
+		public override bool Convert(int i, int j, int conversionType) {
+			if (conversionType == BiomeConversionID.Purity || conversionType == BiomeConversionID.Sand || conversionType == BiomeConversionID.PurificationPowder) {
+				WorldGen.ConvertTile(i, j, TileID.DesertFossil);
+				return false;
+			}
+			else if (conversionType == BiomeConversionID.Corruption) {
+				WorldGen.ConvertTile(i, j, ModContent.TileType<CorruptFossilTile>());
+				return false;
+			}
+			else if (conversionType == BiomeConversionID.Hallow) {
+				WorldGen.ConvertTile(i, j, ModContent.TileType<HallowedFossilTile>());
+				return false;
+			}
+
+			return true;
+		}
+
+		public override void RandomUpdate(int i, int j) {
```

Comment:
Might be worth making a helper method in tML for this kind of thing? That or a common subclass. @JavidPack to weigh in.

Reply:
We're talking about RandomUpdate code right? It's a tough question, maybe a shared static method would be easier for some modders to understand rather than inheritance of a base class. The logic seems like something we can add directly to tmod, and maybe we do want to be able to add a hook in it for allowing additional biome spread limiters that will be honored by modded tiles spreading as well.